test/e2e/framework/framework.go

```diff
@@ -120,7 +120,10 @@ func (f *Framework) WaitForGameServerState(gs *v1alpha1.GameServer, state v1alph
 		return false, nil
 	})
 	if err != nil {
-		return nil, errors.Wrapf(pollErr, "waiting for GameServer to be %v %v/%v: %v",
+		if pollErr != nil {
```

Comment:
Should the line 113 be updated to return the poll error instead? There is no where that PollImmediate is returning err.

Reply:
If we do that, then the retry will stop, and we could get race conditions. We may get an error because the GS isn't created yet (most likely option), but we want to wait for it to show up, and then wait for it to be `Ready`.

My change here allows for if the err != nil (which will happen on timeout), but the pollErr could also be nil, which means you end up with both a nil gameserver and a nil error, and very little idea what happened :smile: 

(This should probably have been it's own PR admittedly. I got a little lazy on this one)

Reply:
I see. Can you please add documentation on why err is replaced with pollErr?

Reply:
So I looked at this code again, and made it much simpler. I don't think we don't need the pollErr  juggling - we have a warn log at that point, so we have visibility, so I got rid of it.

I think this makes the code far cleaner, and easier to read. PTAL :+1: